types/src/transfer.rs

```diff
@@ -168,7 +199,7 @@ impl FromBytes for Transfer {
         let (source, rem) = URef::from_bytes(rem)?;
         let (target, rem) = URef::from_bytes(rem)?;
         let (amount, rem) = U512::from_bytes(rem)?;
-        let (gas, rem) = U512::from_bytes(rem)?;
+        let (_reserved_u512, rem) = U512::from_bytes(rem)?;
```

Comment:
This should be commented to explain why it's required.  Probably `_legacy_gas` might be a better name, as "reserved" implies reserved for future use to me.

Similarly for the two places in the `ToBytes` impl.